### PR TITLE
Update MacOS packaging_games.md

### PR DESCRIPTION
**Pull request by dellis1972**
Update the Packaging instructions for MacOS to include `PublishAot` and other improvements.

### CODE REVIEW

**Comment by mrhelmut** [comment]
Just a note which may be worth turning into a feature in the future: this is the script that we're using to generate automatically a .app bundle when using ```dotnet publish``` https://github.com/FlyingOakGames/GreedyKid/blob/main/GreedyKid_Desktop/GreedyKid_Desktop/post_build.ps1

We can likely do something like that directly built into MonoGame.

**Comment by dellis1972** [comment]
> Just a note which may be worth turning into a feature in the future: this is the script that we're using to generate automatically a .app bundle when using `dotnet publish` https://github.com/FlyingOakGames/GreedyKid/blob/main/GreedyKid_Desktop/GreedyKid_Desktop/post_build.ps1
> 
> We can likely do something like that directly built into MonoGame.
I saw that script. 

I already built a packaging build system a while back via https://www.nuget.org/packages/InfinitespaceStudios.MonoGame.Packaging/#readme-body-tab But that needs to be updated for .net 8 and NativeAOT. Although I'd probably start it from scratch if I had to do that as I'm not sure anyone really used it 🤣 

I was going to submit a "PackageForMac" Target+Task for MonoGame which automated all these steps, but I got the impression that that kind of thing was not wanted in the main repo given that all of the content building msbuild tasks were stripped out in favour of just `Exec` calls. I'm not a fan having to write and maintain .sh or ps1 scripts when users on all platforms will have msbuild available.

**Comment by mrhelmut** [comment]
msbuild sure is the better way to do this and I would vouch for an automatic ```dotnet publish -r osx``` that creates the package by default (although it'd be incomplete because of the x64 and arm64 separation that needs a ```lipo``` to fuse them properly).

It just happened to be a script on our end because that was the most convenient to put together internally, but clearly is sub-optimal.

As for the ```lipo``` step, we do this on Github Actions (because we don't have a mac build box locally to run NativeAOT): https://github.com/FlyingOakGames/PlaceholderGame/blob/2978a0348917192fa8b39c9d3d0798ade40242e5/.github/workflows/build.yml#L70

Basically: we create two .app for x64 and arm64 with NativeAOT, and lipo them post build on Github Actions.

This [placeholdergame](https://github.com/FlyingOakGames/PlaceholderGame) repo basically is how we build our Steam builds for all systems painlessly.

**Review by mrhelmut — DISMISSED** [review]

**Review by SimonDarksideJ — APPROVED** [review]
